pkg/xray/util.go

```diff
@@ -119,6 +119,13 @@ var randomInt = func() func() int {
 	return rand.Int
 }()
 
+func copyStringMap(source map[string]string, target map[string]string) map[string]string {
```

Comment:
you should generalize this as I did with merge maps. It's easy and then it isn't limited to just 2 maps

Reply:
the purpose of this function is just to create a new copy of a single map to avoid modification of the original one. I wasn't planning to merge them somehow.